### PR TITLE
FIX-#419: Handle incompatibility issues between timedelta64 dtypes with Altair

### LINKED ISSUE #419: ValueError: Field "duration" has type "timedelta64[ns]" which is not supported by Altair.
It worked fine with 2 columns of datetimes and 2 columns of integers. I used df.apply(f(x)) to create a timedelta column and got the following warning. Full text:

/opt/conda/lib/python3.7/site-packages/IPython/core/formatters.py:918: UserWarning:
Unexpected error in rendering Lux widget and recommendations. Falling back to Pandas display.
Please report the following issue on Github: https://github.com/lux-org/lux/issues 

/opt/conda/lib/python3.7/site-packages/lux/core/frame.py:632: UserWarning:Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/lux/core/frame.py", line 594, in _ipython_display_
    self.maintain_recs()
  File "/opt/conda/lib/python3.7/site-packages/lux/core/frame.py", line 451, in maintain_recs
    self._widget = rec_df.render_widget()
  File "/opt/conda/lib/python3.7/site-packages/lux/core/frame.py", line 681, in render_widget
    widgetJSON = self.to_JSON(self._rec_info, input_current_vis=input_current_vis)
  File "/opt/conda/lib/python3.7/site-packages/lux/core/frame.py", line 721, in to_JSON
    recCollection = LuxDataFrame.rec_to_JSON(rec_infolist)
  File "/opt/conda/lib/python3.7/site-packages/lux/core/frame.py", line 749, in rec_to_JSON
    chart = vis.to_code(language=lux.config.plotting_backend, prettyOutput=False)
  File "/opt/conda/lib/python3.7/site-packages/lux/vis/Vis.py", line 334, in to_code
    return self.to_vegalite(**kwargs)
  File "/opt/conda/lib/python3.7/site-packages/lux/vis/Vis.py", line 310, in to_vegalite
    self._code = renderer.create_vis(self)
  File "/opt/conda/lib/python3.7/site-packages/lux/vislib/altair/AltairRenderer.py", line 99, in create_vis
    chart_dict = chart.chart.to_dict()
  File "/opt/conda/lib/python3.7/site-packages/altair/vegalite/v4/api.py", line 373, in to_dict
    dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 328, in to_dict
    context=context,
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 62, in _todict
    for k, v in obj.items()
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 63, in <dictcomp>
    if v is not Undefined
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 58, in _todict
    return [_todict(v, validate, context) for v in obj]
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 58, in <listcomp>
    return [_todict(v, validate, context) for v in obj]
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/schemapi.py", line 56, in _todict
    return obj.to_dict(validate=validate, context=context)
  File "/opt/conda/lib/python3.7/site-packages/altair/vegalite/v4/api.py", line 363, in to_dict
    copy.data = _prepare_data(original_data, context)
  File "/opt/conda/lib/python3.7/site-packages/altair/vegalite/v4/api.py", line 84, in _prepare_data
    data = _pipe(data, data_transformers.get())
  File "/opt/conda/lib/python3.7/site-packages/toolz/functoolz.py", line 627, in pipe
    data = func(data)
  File "/opt/conda/lib/python3.7/site-packages/toolz/functoolz.py", line 303, in __call__
    return self._partial(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/altair/vegalite/data.py", line 19, in default_data_transformer
    return curried.pipe(data, limit_rows(max_rows=max_rows), to_values)
  File "/opt/conda/lib/python3.7/site-packages/toolz/functoolz.py", line 627, in pipe
    data = func(data)
  File "/opt/conda/lib/python3.7/site-packages/toolz/functoolz.py", line 303, in __call__
    return self._partial(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/data.py", line 149, in to_values
    data = sanitize_dataframe(data)
  File "/opt/conda/lib/python3.7/site-packages/altair/utils/core.py", line 317, in sanitize_dataframe
    "".format(col_name=col_name, dtype=dtype)
ValueError: Field "duration" has type "timedelta64[ns]" which is not supported by Altair. Please convert to either a timestamp or a numerical value.

### PR DESCRIPTION
**Pull request by cgarciae**
## Overview
Fixes #419 by handling `timedelta64` columns in `sanitize_dataframe`.

## Changes
* Adds tests for the `timedelta64` and `interval` dtypes.
* Converts `timedelta64` columns to `dt.seconds` inside the `sanitize_dataframe` function.

### CODE REVIEW

**Review by dorisjlee — COMMENTED** [review]
This looks good, can we indicate on the visualization labels that the attribute is now "attr [secs]"?

**Comment by cgarciae** [comment]
@dorisjlee what is the best way to do this? Is it just changing the column name or something more fancy?

**Comment by dorisjlee** [comment]
We can do this the same way we set an axis title like we do for the Histogram "axis_name (binned)" for Altair and Matplotlib (see [example source](https://github.com/lux-org/lux/blob/master/lux/vislib/altair/Histogram.py)).

File: lux/vislib/altair/AltairChart.py [hunk 1]
```diff
@@ -123,11 +126,17 @@ def initialize_chart(self):
...
     def sanitize_dataframe(self, df):
+        from lux.utils.date_utils import is_timedelta64_series, timedelta64_to_float_seconds
+
         for attr in df.columns:
             # Check if dtype is unrecognized by Altair (#247)
             if str(df[attr].dtype) == "Float64":
                 df[attr] = df[attr].astype(np.float64)
 
+            # Check for timedelta64[...] dtype
+            if is_timedelta64_series(df[attr]):
```
Comment (cgarciae) [inline lux/vislib/altair/AltairChart.py:137]: @dorisjlee can this be avoided if a preprocess the dataframe in PandasExecutor?

File: lux/executor/PandasExecutor.py [hunk 2]
```diff
@@ -435,7 +439,14 @@ def compute_data_type(self, ldf: LuxDataFrame):
                 ldf._data_type[attr] = ldf._type_override[attr]
             else:
                 temporal_var_list = ["month", "year", "day", "date", "time", "weekday"]
-                if is_datetime(ldf[attr]):
+
+                if is_timedelta64_series(ldf[attr]):
+                    ldf._data_type[attr] = "quantitative"
```
Comment (cgarciae) [inline lux/executor/PandasExecutor.py:444]: @dorisjlee probably here.

Comment (dorisjlee) [reply lux/executor/PandasExecutor.py:444]: converting the data type metadata here is okay, is there a reason why we don't do it in the compiler?

Comment (cgarciae) [reply lux/executor/PandasExecutor.py:444]: I didn't see a place where to do it, if you can point me to it maybe that what we are looking for.

Comment (dorisjlee) [reply lux/executor/PandasExecutor.py:444]: This function is called inside the `Compiler`, so we're all good!

**Review by dorisjlee — APPROVED** [review]

**Pull request merged**